internal/cmd/workspaces.go

```diff
@@ -120,6 +131,201 @@ func lsWorkspacesCommand() *cobra.Command {
 	return cmd
 }
 
+func pingWorkspaceCommand() *cobra.Command {
+	var (
+		schemes []string
+		count   int
+	)
+
+	cmd := &cobra.Command{
+		Use:     "ping [workspace_name]",
+		Short:   "ping Coder workspaces by name",
+		Long:    "ping Coder workspaces by name",
+		Example: `coder workspaces ping front-end-workspace`,
+		Args:    xcobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			client, err := newClient(ctx, true)
+			if err != nil {
+				return err
+			}
+			workspace, err := findWorkspace(ctx, client, args[0], coder.Me)
+			if err != nil {
+				return err
+			}
+
+			iceSchemes := map[ice.SchemeType]interface{}{}
+			for _, rawScheme := range schemes {
+				scheme := ice.NewSchemeType(rawScheme)
+				if scheme == ice.Unknown {
+					return fmt.Errorf("scheme type %q not recognized", rawScheme)
+				}
+				iceSchemes[scheme] = nil
+			}
+
+			pinger := &wsPinger{
+				client:     client,
+				workspace:  workspace,
+				iceSchemes: iceSchemes,
+			}
+
+			seq := 0
+			ticker := time.NewTicker(time.Second)
+			for {
+				select {
+				case <-ticker.C:
+					err := pinger.ping(ctx)
+					if err != nil {
+						return err
+					}
+					seq++
+					if count != 0 && seq >= count {
+						os.Exit(0)
+					}
+				case <-ctx.Done():
+					return nil
+				}
+			}
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&schemes, "scheme", "s", []string{"stun", "stuns", "turn", "turns"}, "customize schemes to filter ice servers")
+	cmd.Flags().IntVarP(&count, "count", "c", 0, "stop after <count> replies")
+	return cmd
+}
+
+type wsPinger struct {
+	client     coder.Client
+	workspace  *coder.Workspace
+	dialer     *wsnet.Dialer
+	iceSchemes map[ice.SchemeType]interface{}
+	tunneled   bool
+}
+
+// Only return fatal errors
+func (w *wsPinger) ping(ctx context.Context) error {
+	ctx, cancelFunc := context.WithTimeout(ctx, time.Second*15)
+	defer cancelFunc()
+	url := w.client.BaseURL()
+	logFail := func(msg string) {
+		fmt.Printf("%s: %s\n", color.New(color.Bold, color.FgRed).Sprint("——"), msg)
+	}
+	logSuccess := func(timeStr, msg string) {
+		fmt.Printf("%s: %s\n", color.New(color.Bold, color.FgGreen).Sprint(timeStr), msg)
+	}
+
+	// If the dialer is nil we create a new!
+	// nolint:nestif
+	if w.dialer == nil {
+		servers, err := w.client.ICEServers(ctx)
+		if err != nil {
+			logFail(fmt.Sprintf("list ice servers: %s", err.Error()))
+			return nil
+		}
+		filteredServers := make([]webrtc.ICEServer, 0)
```

Comment:
minor minor thing but we could preallocate this since we know the maximum possible length is `len(servers)`

```suggestion
		filteredServers := make([]webrtc.ICEServer, 0, len(servers))
```